src/bloqade/pyqrack/squin/qubit.py

```diff
@@ -0,0 +1,111 @@
+from typing import Any
+
+from kirin import interp
+from kirin.dialects import ilist
+from kirin.interp.exceptions import InterpreterError
+
+from bloqade.squin import qubit
+from bloqade.pyqrack.reg import QubitState, PyQrackQubit
+from bloqade.pyqrack.base import PyQrackInterpreter
+
+from .runtime import OperatorRuntimeABC
+
+
+@qubit.dialect.register(key="pyqrack")
+class PyQrackMethods(interp.MethodTable):
+    @interp.impl(qubit.New)
+    def new(self, interp: PyQrackInterpreter, frame: interp.Frame, stmt: qubit.New):
+        n_qubits: int = frame.get(stmt.n_qubits)
+        qreg = ilist.IList(
+            [
+                PyQrackQubit(i, interp.memory.sim_reg, QubitState.Active)
+                for i in interp.memory.allocate(n_qubits=n_qubits)
+            ]
+        )
+        return (qreg,)
+
+    @interp.impl(qubit.Apply)
+    def apply(self, interp: PyQrackInterpreter, frame: interp.Frame, stmt: qubit.Apply):
+        qubits: ilist.IList[PyQrackQubit, Any] = frame.get(stmt.qubits)
+
+        for qbit in qubits:
+            if not qbit.is_active():
+                return ()
```

Comment:
Let's leave this logic to the `operator.apply`

Reply:
@weinbe58 I thought about this. It's either checking once here, or in every `apply` method in the runtime. Are you sure we'd want that? Of course, I can also make it an abstract method that checks this and then calls e.g. `_apply`, but that's also not so pretty.

Reply:
Never mind, as discussed below, we can have situations where there are qubits passed in that have nothing to do with the gate that is being applied and we should just ignore those.